src/xtopcom/xstatistic/xstatistic.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2017-2020 Telos Foundation & contributors
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#pragma once
+
+#include "xbase/xdata.h"
+#include "xbasic/xmemory.hpp"
+
+#include <atomic>
+#include <string>
+
+// #define USE_MULTISET_ONLY
+
+NS_BEG2(top, xstatistic)
+
+enum enum_statistic_class_type {
+    enum_statistic_begin = 0,
+    enum_statistic_tx_v2 = enum_statistic_begin,
+    enum_statistic_tx_v3,
+    enum_statistic_receipt,
+    enum_statistic_vqcert,
+    enum_statistic_vblock,
+    enum_statistic_table_bstate,
+    enum_statistic_unit_bstate,
+    enum_statistic_block_header,
+    enum_statistic_vinput,
+    enum_statistic_voutput,
+    enum_statistic_vbstate,
+    enum_statistic_vcanvas,
+    enum_statistic_max,
+    enum_statistic_invalid = enum_statistic_max,

```

Comment:
invalid enum value is recommended to be the default value which is zero, thus the use side can simply use enum_statictic_class_type{} to init to the default invalid value.

Reply:
it's the same rule for all types. init to ZERO is the default behavior which is better for representing INVALID state (if exists) by default.

Reply:
those enum values will be used for vector index, start from 0 can be more handy.